third_party_integration/huggingface_transformers/BootstrapNAS_HF/0001-Modifications-for-BootstrapNAS-usage.patch

```diff
@@ -1216,33 +1378,16 @@ index 214e7a978..75c139923 100755
              self._signature_columns = list(signature.parameters.keys())
              # Labels may be named label or label_ids, the default data collator handles that.
              self._signature_columns += list(set(["label", "label_ids"] + self.label_names))
-@@ -1146,6 +1160,25 @@ class Trainer:
+@@ -1144,6 +1158,8 @@ class Trainer:
+                 num_warmup_steps=self.args.get_warmup_steps(num_training_steps),
+                 num_training_steps=num_training_steps,
              )
++        if self.compression_ctrl is not None:
++            self.compression_ctrl.scheduler.lr_scheduler = self.lr_scheduler
```

Comment:
We must link self.lr_scheduler with our self.compression_ctrl.scheduler.lr_scheduler. Because we need [lr_scheduler.current_step](https://github.com/openvinotoolkit/nncf/blob/2c5ae0919cf714062ca39979235a0e2e1adb666e/nncf/experimental/torch/nas/bootstrapNAS/training/scheduler.py#L135) to [activate random subnets](https://github.com/openvinotoolkit/nncf/blob/2c5ae0919cf714062ca39979235a0e2e1adb666e/nncf/experimental/torch/nas/bootstrapNAS/training/progressive_shrinking_controller.py#L138).